pkg/k8scontext/k8scontext_test.go

```diff
@@ -0,0 +1,78 @@
+package k8scontext_test
+
+import (
+	go_flag "flag"
+	"time"
+
+	"github.com/Azure/application-gateway-kubernetes-ingress/pkg/k8scontext"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/api/extensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/kubernetes"
+	testclient "k8s.io/client-go/kubernetes/fake"
+)
+
+var _ = Describe("K8scontext", func() {
+	var k8sClient kubernetes.Interface
+	var ctxt *k8scontext.Context
+	ingressNS := "test-ingress-controller"
+
+	go_flag.Lookup("logtostderr").Value.Set("true")
+	go_flag.Set("v", "3")
+
+	BeforeEach(func() {
+		// Create the mock K8s client.
+		k8sClient = testclient.NewSimpleClientset()
+
+		// Create the "test-ingress-controller" namespace.
+		// We will create all our resources under this namespace.
+		ns := &v1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: ingressNS,
+			},
+		}
+		_, err := k8sClient.CoreV1().Namespaces().Create(ns)
+
+		Expect(err).Should(BeNil(), "Unable to create the mock Kuberentes client: %v", err)
```

Comment:
nit: typo kuberentes and should the message be more like "unable to create namespace"